### ISSUE TITLE
Version 0.12.3 x86 and x64 Windows installer possibly corrupted.

### ISSUE THREAD

**Issue by TheNephalim** [joyent/node#25347]
Hello,

I believe that both the 32- and 64-bit MSI installers for Node version 0.12.3 are corrupted.  Installation attempts result in the following error:

"This installation package could not be opened.  Contact the application vendor to verify that this is a valid windows installer package."

Common resolutions for this error failed.  I downloaded the version 0.12.2 version and could successfully install that version.  I currently run 64-bit Windows 7 Enterprise with Service Pack 1 on my desktop.

- Robert

**Comment by deliverymanager** [comment]
I have a similar problem with WIN 32 installation version 0.12.3 x86.
After successfully installing on a Windows XP platform, then when I try to open the Node.exe it says that " ... is not a valid Win32 application"
I also, uninstalled it and then installed version 0.10.33 and it worked fine.
Then I downloaded again the msi and reinstalled the latest 0.12.3 and the same error appeared.

**Comment by deliverymanager** [comment]
Also I have installed the version 0.12.2 x86 and it is working fine on Windows XP.
The only version that gives the not valid win32 application, is the version 0.12.3 
I have also tested on a Windows 7 platform win32 and the version 0.12.3 is working fine.
So I think the problem is related to the Windows XP specific platform.
I think I am gonna open a new issue for this specific problem on Windows XP with the latest version.

**Comment by misterdjules** [comment]
@orangemocha Do you have some time to investigate?

**Comment by orangemocha** [comment]
Yes, @joaocgreis has already gotten to the bottom of it.

**Comment by joaocgreis** [comment]
Hello @TheNephalim

I was able to install both 32- and 64-bit versions on my Windows 7 Enterprise 64-bit machine.

Can you make sure the MSI file is correct by verifying the SHA1 sum? It should be as listed in http://nodejs.org/dist/v0.12.3/SHASUMS.txt.asc . You can use a tool like http://code.kliu.org/hashcheck/ for this. Please download again if it is not correct.

Make sure you unblock the file before trying to install by going to right click, properties, and clicking the unblock button at the bottom.

**Comment by orangemocha** [comment]
@deliverymanager with version 0.12 support for XP has been deprecated.

**Comment by deliverymanager** [comment]
Really? Is Windows XP support deprecated?
Could you please point me to the release notes or change log that says that?
I am really really disappointed... I know that Windows Xp is really really old and not supported by Microsoft any more, but it is really important for me to be able to run Node on Windows XP.
So the only solution for me would be to run an old version of node?

**Comment by misterdjules** [comment]
@orangemocha @joaocgreis It seems that node v0.12.2 [can be installed on Windows XP](https://github.com/joyent/node/issues/25347#issuecomment-103136154), so I would expect v0.12.3 to install and run properly on Windows XP too.

I can't think of anything that changed between v0.12.2 and v0.12.3 that could cause such a regression (if the regression is confirmed). Maybe we need to backport https://github.com/nodejs/io.js/commit/e8d08503c7821e8c92e9fa236ed7328e9bdfe62a?

**Comment by deliverymanager** [comment]
By my side as I already wrote I can confirm that 0.12.2 is actually installed, running and working properly on Win XP.
0.12.3 is not for sure after making several installations on clean computer.
Hope this can be resolved.

**Comment by orangemocha** [comment]
@deliverymanager can you check to make sure that your download is not corrupt, as suggested by @joaocgreis ?

>Can you make sure the MSI file is correct by verifying the SHA1 sum? It should be as listed in http://nodejs.org/dist/v0.12.3/SHASUMS.txt.asc . You can use a tool like http://code.kliu.org/hashcheck/ for this. Please download again if it is not correct.

>Make sure you unblock the file before trying to install by going to right click, properties, and clicking the unblock button at the bottom.

Whether or not XP is officiallly supported, "is not a valid Win32 application" looks like a corrupted download.

I'll follow up separately on the issue of XP support. It's a bigger decision and it should be discussed with the TSC. You are right that right now it's not documented properly.

**Comment by deliverymanager** [comment]
I will check if the file is corrupted and I will let you know.

**Comment by orangemocha** [comment]
Re-opened https://github.com/joyent/node/issues/25348 to track the Windows XP specific issue.

**Comment by TheNephalim** [comment]
@joaocgreis I can confirm that the error message I got during the installation process stated that the SHA1 sums were not correct.  I attempted downloading and re-downloading the files and received the same results.  I did ensure that I unblocked the MSI file downloaded from the site before installation.  Also ensured the Windows Installer Service was started, unregistered and re-registered the msiexec.exe for both x86 and x64 bit, and nothing worked.  So...it is possible that something could have been corrupting the files during the download process.  I do have one question:  does the installation process require downloading additional information from any site requiring https?  My work utilizes proxies and this has caused me problems before with Ruby, etc.

**Comment by misterdjules** [comment]
@TheNephalim Any update on your investigations?

**Comment by TheNephalim** [comment]
@misterdjules I apologize for the very long delay in posting results.

I do not get a matches in the checksums for v0.12.3 and v0.12.4.  I do get matches for the checksums for v0.12.1 and v0.12.2.  Included below are the results from the files I downloaded.

node-v0.12.4-x64.msi - Does not match

SHA-1       -  B1B713418B28993AFD10FC228875D84D1EDD5C5E
SHA-256   -  5A6EAEF0BFB3462F6913108EABA4466FA98D0764A158E3C5E184CC8783ECE29F
SHA-512   -    93ECB4C6385DE95A42340A360198F65EB930EEC97B33F5371F1649AA109AEBACCDAB121A50F6D2E754D55C75691A5D4C09BB5F3F1403F06207957B844EBF6FAE

node-v0.12.3-x64,msi - Does not match

SHA-1       -  292360BD318398CAE80D681EC179FBCE0B2ED1BC
SHA-256   -  0A3ECB44827817D9E04A0118792DFF25173758400BC10C5B639A8462FBCFA983
SHA-512 -   A13F5054C73227B25D0F0B1E330637EAA3990D0223EAB666E0BAE0755CC52693B1AD598C6F25FEF0DC40B9B83873920ABBC0636CED49697672A7D75AB75A071D

node-v0.12.2-x64.msi - Match

SHA-1     - 994A1461DF5DD7DE79F8072B14103642D0D5CBFE
SHA-256 - 898F9AA7155626C974035A238194EA75E3561D30D09ED12A65461804BF4AAB58
SHA-512 - 750DAE553273D8A8BDA046F4F31AE9CCAD054B3F5BF8FD5EFBC01B84C081869D834C4575292E6D88BAC978BA3A7FD6FCEC0B3B95712D66737E1627FF4A5C3D81

node-v0.12.1-x64 - Match

SHA-1     - 2BA47A75E8F8CF118C0564641449BFACE77ED07F
SHA-256 - BE8CB6F1FC2A98D2A330DADA359B47CC10F6709AF7939BA36E9C3C9B5B61E0E8
SHA-512 -  92183A0E9CF753C43CE5E628D9132ABE5CC334A2C03D563606D91CC314A65C80973099FC1E3208A49C2470B0860E21614FE899222212235C4368582C5C0DEF2C

**Comment by misterdjules** [comment]
@TheNephalim What do you use to compute the sha1 of x64 MSIs? Here's what I get:
```
➜  v0.12 git:(v0.12) ✗ curl -s http://nodejs.org/dist/v0.12.4/x64/node-v0.12.4-x64.msi | shasum
69f15439af92c5e0beec02dc8145b74d98c103dd  -
➜  v0.12 git:(v0.12) ✗ curl -s http://nodejs.org/dist/v0.12.3/x64/node-v0.12.3-x64.msi | shasum
3677034221eeb300d7cd0cf2aaf3d172ab05fd11  -
➜  v0.12 git:(v0.12) ✗
```
Both results match what's in the `SHASUMS` files on nodejs.org.

**Comment by TheNephalim** [comment]
@misterdjules I used the following:  MD5_and_SHA_Checksum_Utility and HashCheck Shell Extension (recommended by @joaocgreis ).  The HashCheck Shell extension utility only does Sha-1, but it matches what I got from the other utility.  Here are the results from the HashCheck Shell Extension utility:

292360bd318398cae80d681ec179fbce0b2ed1bc *node-v0.12.3-x64.msi
b1b713418b28993afd10fc228875d84d1edd5c5e *node-v0.12.4-x64.msi

I noticed that you are using curl to retrieve the files and calculate the hash.  I can try to download and do the same to see what the results are, but it is weird that up until 0.12.2 the hashes match what is on the site and 0.12.3 and beyond do not match.

**Comment by misterdjules** [comment]
@TheNephalim I used the HashCheck shell extension on Windows 7 to check the sha1 checksums of both node-v0.12.3-x64.msi and node-v0.12.4-x64.msi, and I found them to match what's in nodejs.org/dist/.

You mentioned in an earlier comment that you were downloading files behind a proxy, have you been able to download these files while not behind that proxy?

**Comment by TheNephalim** [comment]
@misterdjules Not yet.  Will try it out.

**Comment by TheNephalim** [comment]
@misterdjules I didn't check the checksums, but I did download node-v0.12.4-x64.msi and it installed without issue on my home computer running Windows 7 Ultimate.  So, there must be something in the proxy that is corrupting the download of these files.  It is weird that I am able to download and install versions below v0.12.3 without issue.  I guess the resolution for me is to download the MSIs at home and then bring in to install.  Unless you have some other potential work around.

**Comment by misterdjules** [comment]
@TheNephalim It seems this issue could be related to https://github.com/joyent/node/issues/25531, in the sense that that issue also deals with a proxy that seems to act as some sort of man in the middle to alter the content of what is transmitted.

We recently changed all download links on nodejs.org to use HTTPS, and that should prevent proxies from altering the content that is sent on the wire. Could you please try again to download node-v0.12.4-x64.msi from behind your proxy and let us know if the switch to HTTPS fixed this issue?

Thank you for your help :+1:

**Comment by TheNephalim** [comment]
@misterdjules I downloaded node-v0.12.5-x64.msi from https://nodejs.org and it was a no-go.  I can confirm that my company does a man-in-the-middle insertion of certificates.  For example, I can look at the https connection to https://github.org and I have 3 certificates from my company before you get to the github.com certificates.  

I understand it, to a degree, from a security perspective, but it is a massive PITA when dealing with nodejs and ruby.

I don't know how much help I've been to you, but I appreciate you at least taking a look at this problem and trying to resolve it.

**Comment by misterdjules** [comment]
@TheNephalim If you have the certificate/CA of the SSL proxy in your browser's list of trusted certificates/CAs, then I can't think of any way we might be able to help you. It seems that this issue is specific to your setup, so I'm closing it but feel free to suggest any better way to handle this.

Thank you for helping us troubleshot this issue, and let us know if we can do anything more to help! :+1:
